### PR TITLE
JSDoc: Logging

### PR DESCRIPTION
**Pull request by mischnic**
@DeMoorJasper I've copied the text from your docs page. Not sure about `skipFormatting` though...

Preview:
https://website-git-api-logging.devongovett.vercel.app/plugin-system/logging/
https://github.com/parcel-bundler/website/pull/649

### CODE REVIEW

File: packages/core/diagnostic/src/diagnostic.js [hunk 1]
```diff
@@ -4,49 +4,71 @@ import type {FilePath} from '@parcel/types';
...
-  // Codeframe data
+  /** A code frame points to a certain location(s) in the file this diagnostic is linked to (optional) */
   codeFrame?: DiagnosticCodeFrame,
 
-  // Hints to resolve issues faster
+  /** An optional list of strings that suggest ways to resolve this issue */
   hints?: Array<string>,
 
+  /** TODO? */
   skipFormatting?: boolean,
```
Comment (mischnic) [inline packages/core/diagnostic/src/diagnostic.js:72]: ...

File: packages/core/diagnostic/src/diagnostic.js [hunk 2]
```diff
@@ -141,11 +165,12 @@ export function errorToDiagnostic(
...
 }
 
-type ThrowableDiagnosticOpts = {
+export type ThrowableDiagnosticOpts = {
   diagnostic: Diagnostic | Array<Diagnostic>,
   ...
 };
 
+// TODO
 export default class ThrowableDiagnostic extends Error {
```
Comment (mischnic) [inline packages/core/diagnostic/src/diagnostic.js:174]: ...

File: packages/core/diagnostic/src/diagnostic.js [hunk 3]
```diff
@@ -179,7 +211,12 @@ export function generateJSONCodeHighlights(
   });
 }
 
+/**
+ * Converts entries in <a href="https://github.com/epoberezkin/json-source-map">epoberezkin/json-source-map</a>'s
+ * <code>result.pointers</code> array.
+ */
 export function getJSONSourceLocation(
+  // TODO: remove `any` here
```
Comment (mischnic) [inline packages/core/diagnostic/src/diagnostic.js:219]: ...

File: packages/core/diagnostic/src/diagnostic.js (hunk 1 above)
Comment (DeMoorJasper) [reply packages/core/diagnostic/src/diagnostic.js:72]: This should actually be hidden in the docs, we only really use it to not format things that weren't originally designed for diagnostics

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 7.46s | +26.00ms |
| Cached | 2.64s | +22.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.d5807e82.webp | 102.94kb | +0.00b | 294.00ms | +28.00ms ⚠️ |
| dist/modern/parcel.d5807e82.webp | 102.94kb | +0.00b | 295.00ms | +30.00ms ⚠️ |

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 34.55s | -1.76s 🚀 |
| Cached | 2.88s | +41.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 187.00ms | -26.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.js | 478.68kb | +0.00b | 421.00ms | +85.00ms ⚠️ |
| dist/PermalinkedComment.9c5ad065.js | 4.22kb | +0.00b | 417.00ms | +84.00ms ⚠️ |
| dist/UserProfile.c1927888.js | 1.62kb | +0.00b | 417.00ms | +84.00ms ⚠️ |
| dist/NotFound.f8a0616d.js | 387.00b | +0.00b | 417.00ms | +84.00ms ⚠️ |
| dist/logo.24c8bf9e.png | 274.00b | +0.00b | 311.00ms | +47.00ms ⚠️ |

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 3.50m | +2.86s |
| Cached | 6.25s | +129.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/popup.6c453e3c.js | 169.96kb | +0.00b | 37.62s | -27.71s 🚀 |
| dist/EmojiPickerComponent.ae601247.js | 75.29kb | +0.00b | 11.90s | -646.00ms 🚀 |
| dist/media-viewer.bcba38b3.js | 71.96kb | +0.00b | 16.51s | -950.00ms 🚀 |
| dist/card.645a5c17.js | 55.01kb | +0.00b | 13.95s | -709.00ms 🚀 |
| dist/card.0313bac8.js | 52.78kb | +0.00b | 1.10m | -44.13s 🚀 |
| dist/DatePicker.0df05dac.js | 21.21kb | +0.00b | 11.90s | -646.00ms 🚀 |
| dist/EmojiPickerComponent.7dc0ebd0.js | 18.23kb | +0.00b | 11.90s | -647.00ms 🚀 |
| dist/smartMediaEditor.9623570c.js | 16.87kb | +0.00b | 1.83m | +42.13s ⚠️ |
| dist/component.77c0f682.js | 6.05kb | +0.00b | 6.64s | +647.00ms ⚠️ |
| dist/media-viewer.59c17dad.js | 3.92kb | +0.00b | 16.50s | -949.00ms 🚀 |
| dist/EmojiPickerComponent.627aca6b.js | 3.58kb | +0.00b | 11.90s | -646.00ms 🚀 |
| dist/card.eea6ad8c.js | 2.03kb | +0.00b | 16.51s | -947.00ms 🚀 |
| dist/ResourcedEmojiComponent.abe8596c.js | 2.00kb | +0.00b | 6.32s | +518.00ms ⚠️ |
| dist/16.1b6ca79f.js | 1.66kb | +0.00b | 8.69s | -488.00ms 🚀 |
| dist/16.19c93285.js | 1.33kb | +0.00b | 8.62s | -560.00ms 🚀 |
| dist/16.9ba230ea.js | 1.13kb | +0.00b | 9.13s | -624.00ms 🚀 |
| dist/16.eafe1a11.js | 1.13kb | +0.00b | 8.62s | -560.00ms 🚀 |
| dist/16.3920766f.js | 1.10kb | +0.00b | 6.32s | +519.00ms ⚠️ |
| dist/component.6600f83e.js | 1.07kb | +0.00b | 6.32s | +518.00ms ⚠️ |
| dist/media-viewer-analytics-error-boundary.0afa7031.js | 755.00b | +0.00b | 16.50s | -949.00ms 🚀 |
| dist/EmojiPickerComponent.2e54b9c6.js | 535.00b | +0.00b | 11.90s | -647.00ms 🚀 |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/index.2180c879.js | 2.72mb | +2.00b ⚠️ | 649.00ms | +33.00ms ⚠️ |
| dist/pdfRenderer.15a6a023.js | 1.11mb | -6.00b 🚀 | 636.00ms | +31.00ms |
| dist/editorView.e06544fb.js | 587.12kb | +0.00b | 636.00ms | +32.00ms ⚠️ |
| dist/Toolbar.83ac0b6e.js | 92.34kb | +0.00b | 636.00ms | +32.00ms ⚠️ |
| dist/Modal.19ffed32.js | 41.27kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/component.d7c29980.js | 30.70kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/component.54080a26.js | 22.50kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/smartMediaEditor.9623570c.js | 16.87kb | +0.00b | 636.00ms | +32.00ms ⚠️ |
| dist/js.5733a017.js | 16.40kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/component.77c0f682.js | 6.05kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/index.ac562c80.css | 3.46kb | +0.00b | 636.00ms | +32.00ms ⚠️ |
| dist/png-chunks-extract.a844fdb6.js | 3.41kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/Modal.0304b9ed.js | 3.08kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/16.290b4a43.js | 2.38kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/ResourcedEmojiComponent.abe8596c.js | 2.00kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/date.231500d2.js | 1.73kb | +0.00b | 646.00ms | +37.00ms ⚠️ |
| dist/16.d16f14ed.js | 1.73kb | +0.00b | 647.00ms | +38.00ms ⚠️ |
| dist/images.80c92f5e.js | 1.67kb | +0.00b | 647.00ms | +38.00ms ⚠️ |
| dist/16.1b6ca79f.js | 1.66kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/list-number.c35f9589.js | 1.46kb | +0.00b | 646.00ms | +38.00ms ⚠️ |
| dist/16.9ccb5b28.js | 1.38kb | +0.00b | 647.00ms | +38.00ms ⚠️ |
| dist/code.3dd71280.js | 1.38kb | +0.00b | 646.00ms | +37.00ms ⚠️ |
| dist/16.19c93285.js | 1.33kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/16.c7751e34.js | 1.33kb | +0.00b | 647.00ms | +37.00ms ⚠️ |
| dist/16.51049e2c.js | 1.32kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/16.18344f47.js | 1.31kb | +0.00b | 646.00ms | +37.00ms ⚠️ |
| dist/link.2f4427e6.js | 1.30kb | +0.00b | 646.00ms | +38.00ms ⚠️ |
| dist/16.2c4c1846.js | 1.27kb | +0.00b | 647.00ms | +38.00ms ⚠️ |
| dist/emoji.93ae61d3.js | 1.23kb | +0.00b | 647.00ms | +38.00ms ⚠️ |
| dist/16.d17ab186.js | 1.23kb | +0.00b | 647.00ms | +38.00ms ⚠️ |
| dist/16.436f097f.js | 1.21kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/16.b4da0acb.js | 1.19kb | +0.00b | 647.00ms | +38.00ms ⚠️ |
| dist/16.02c68c4f.js | 1.18kb | +0.00b | 646.00ms | +37.00ms ⚠️ |
| dist/16.b46ef690.js | 1.14kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/16.4e3b0235.js | 1.14kb | +0.00b | 647.00ms | +38.00ms ⚠️ |
| dist/16.9ba230ea.js | 1.13kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/16.a37df53b.js | 1.13kb | +0.00b | 647.00ms | +38.00ms ⚠️ |
| dist/16.9adb4ec7.js | 1.13kb | +0.00b | 647.00ms | +38.00ms ⚠️ |
| dist/16.eafe1a11.js | 1.13kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/16.3920766f.js | 1.10kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/component.6600f83e.js | 1.07kb | +0.00b | 648.00ms | +38.00ms ⚠️ |
| dist/mention.4d64e059.js | 1.07kb | +0.00b | 646.00ms | +40.00ms ⚠️ |
| dist/16.afcb09b6.js | 1.05kb | +0.00b | 647.00ms | +38.00ms ⚠️ |
| dist/layout.52ba15f7.js | 1.05kb | +0.00b | 647.00ms | +38.00ms ⚠️ |
| dist/divider.16defde6.js | 1.03kb | +0.00b | 646.00ms | +38.00ms ⚠️ |
| dist/action.9aa72b39.js | 1.00kb | +0.00b | 646.00ms | +37.00ms ⚠️ |
| dist/decision.20244046.js | 1010.00b | +0.00b | 646.00ms | +38.00ms ⚠️ |
| dist/list.c9147a62.js | 978.00b | +0.00b | 646.00ms | +40.00ms ⚠️ |
| dist/panel-error.8d11e8b7.js | 906.00b | +0.00b | 646.00ms | +40.00ms ⚠️ |
| dist/index.html | 119.00b | +0.00b | 651.00ms | +33.00ms ⚠️ |

</p></details>

<details><summary>Three.js x4 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/275292203763368461)**

**Review by DeMoorJasper — COMMENTED** [review]
Looks good to me.

I can write the docs for ThrowableDiagnostic in my diagnostic improvements PR if you want once this one's merged.

**Review by DeMoorJasper — APPROVED** [review]

**Comment by mischnic** [comment]
I didn't see your comment 😄

**Pull request merged**